### ISSUE TITLE
Block inside if

### ISSUE THREAD

**Issue by wallneradam** [pugjs/pug#2586]
I have a layout like this:

```
if !layout_included
    - var layout_included = true

    doctype html

    body
        p
            block content

else
    block content
```

And a foo.pug file using it:

```
extends layout

block content
    strong Test
```

And maybe another using this included in:

```
extends layout

block content
    p Blah blah
    include foo
```

The use case is that I can use the same file either as an include or a stand alone view. It was working very well in Jade, but in pug the layout simply put the content in the last occurence of block section of the file in my case the one in "else".

I think it is a bug.

**Comment by wallneradam** [comment]
It seems to be a duplicate of this bug: #2367

I've tried this workaround:
```
mixin theBlock
    block content

if !layout_included
    doctype html
    body
        p --- Before content ---
        +theBlock
        p -- After content ---

else
    +theBlock

- var layout_included = true;
```

But this does not work as well, because the second call of extends doesn't have the variable layout_included defined. It worked in Jade as well.

**Comment by wallneradam** [comment]
Ok the final workaround is using locals:

```
mixin theBlock
    block content

if !locals.layout_included
    - locals.layout_included = true;

    doctype html
    body
        p --- Before content ---
        +theBlock
        p -- After content ---

else
    +theBlock
```

**Comment by matheusgrieger** [comment]
This workaround from @wallneradam worked perfectly for me.

**Comment by Tiliavir** [comment]
should be closed as duplicate of #2367
